pages/weather.js

```diff
@@ -1,3 +1,244 @@
+import React, { useEffect, useState } from "react";
+import useLocalStorageState from "use-local-storage-state";
+import useSWR from "swr";
+import Image from "next/image";
+import { StyledSection, StyledRowSection } from "@/components/StyledSections";
+import { Label, Input } from "@/components/StyledFormElements";
+import styled from "styled-components";
+import { StyledSearchButton } from "@/components/StyledButtons";
+
+const StyledWeatherDay = styled.div`
+  border: 1px solid grey;
+  padding: 0.3rem;
+  margin-bottom: 1rem;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 21rem;
+`;
+
+const fetcher = async (url) => {
+  const res = await fetch(url);
+
+  if (!res.ok) {
+    const error = new Error("An error occurred while fetching the data.");
+    error.info = await res.json();
+    error.status = res.status;
+    throw error;
+  }
+
+  return res.json();
+};
+
 export default function WeatherPage() {
-  return <h1>Weather page</h1>;
+  const [weatherLocation, setWeatherLocation] = useLocalStorageState(
+    "weatherLocation",
+    { defaultValue: "Hamburg" }
+  );
+  const [url, setUrl] = useState(
+    `https://api.weatherapi.com/v1/forecast.json?key=a7f12ff76d6c4b7383f102231231505&q=${weatherLocation}&days=7&aqi=no&alerts=no`
+  );
+
+  const { data, error, isLoading } = useSWR(url, fetcher);
+
+  useEffect(() => {
+    setUrl(
+      `https://api.weatherapi.com/v1/forecast.json?key=a7f12ff76d6c4b7383f102231231505&q=${weatherLocation}&days=7&aqi=no&alerts=no`
+    );
+  }, [weatherLocation]);
+
+  if (error) return <div>failed to load</div>;
+  if (isLoading) return <div>loading...</div>;
+
+  const currentDate = new Date();
+  const tomorrowDate = new Date(currentDate);
+  tomorrowDate.setDate(currentDate.getDate() + 1);
+  const options = { weekday: "long" };
+  const todayWeekday = currentDate.toLocaleDateString("en-US", options);
+  const tomorrowWeekday = tomorrowDate.toLocaleDateString("en-US", options);
+
+  function handleKeyPress(event) {
+    if (event.key === "Enter") {
+      event.preventDefault();
+      const newLocation = event.target.value;
+      setWeatherLocation(newLocation);
+    }
+  }
```

Comment:
you don't need this function, when hitting "enter" the form is submitted and the handleLocationSubmit is called.

You would set the state twice.

```suggestion
```